### PR TITLE
Bip39 mnemonic verification

### LINKED ISSUE #1895: Add endpoint to verify bip39 seed

### PR DESCRIPTION
**Pull request by J0nathan-S**
Fixes #1895

Changes:
- Added Bip39 mnemonic verification

Does this change need to mentioned in CHANGELOG.md?
yes

### CODE REVIEW

File: src/api/wallet_test.go [hunk 1]
```diff
@@ -1693,6 +1694,78 @@ func TestWalletNewSeed(t *testing.T) {
...
+			status: http.StatusBadRequest,
+			err:    fmt.Errorf("400 Bad Request - missing seed"),
+		},
+		{
+			method: http.MethodGet,
+			status: http.StatusOK,
+			seed:   "chief stadium sniff exhibit ostrich exit fruit noodle good lava coin supply",
+			err:    nil,
+		},
+	}
```
Comment (therealssj) [inline src/api/wallet_test.go]: wrong tests. the method is actually post

File: src/api/wallet.go [hunk 2]
```diff
@@ -790,16 +788,73 @@ func walletSeedHandler(gateway Gatewayer) http.HandlerFunc {
...
+
+		if mn.Seed == "" {
+			resp := NewHTTPErrorResponse(http.StatusBadRequest, "missing seed")
+			writeHTTPResponse(w, resp)
+			return
+		}
+
+		mnemonicValid := bip39.IsMnemonicValid(mn.Seed)
+		if !mnemonicValid {
+			resp := NewHTTPErrorResponse(http.StatusBadRequest, "seed is not a valid bip39 seed")
```
Comment (gz-c) [inline src/api/wallet.go]: Use `StatusUnprocessableEntity`, same as `/api/v2/address/verify`, and update the api/README.md entry similarly

File: src/api/wallet_test.go [hunk 3]
```diff
@@ -1693,6 +1694,86 @@ func TestWalletNewSeed(t *testing.T) {
...
+	tt := []struct {
+		method string
+		status int
+		seed   string
+		err    error
+	}{
+		{
+			method: http.MethodGet,
+			status: http.StatusMethodNotAllowed,
+			err:    fmt.Errorf(http.StatusText(http.StatusMethodNotAllowed)),
```
Comment (gz-c) [inline src/api/wallet_test.go]: you can use `errors.New` if not using a format string (one with `%` in it)

File: src/api/wallet_test.go [hunk 4]
```diff
@@ -1693,6 +1694,86 @@ func TestWalletNewSeed(t *testing.T) {
...
+	}
+
+	for _, tc := range tt {
+		gateway := &MockGatewayer{}
+		endpoint := "/api/v2/wallet/seed/verify"
+
+		v := SeedVerificationReq{Seed: tc.seed}
+
+		body, err := json.Marshal(v)
+		if err != nil {
```
Comment (gz-c) [inline src/api/wallet_test.go]: use `require.NoError`, don't use the `t.Error*` methods to fail tests

**Comment by gz-c** [comment]
The `"valid"` field in the response can be removed, it is not necessary. The `200 OK` response indicates that it is valid

**Pull request closed without merging**